pkg/webhook/common_test.go

```diff
@@ -0,0 +1,298 @@
+package webhook
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"math/big"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+func TestCongifureWebhookServer(t *testing.T) {
+	expectedServer := &webhook.Server{
+		TLSMinVersion: "1.3",
+	}
+
+	if *clientCAName != "" {
+		expectedServer.ClientCAName = *clientCAName
+	}
+
+	tc := []struct {
+		Name           string
+		Server         *webhook.Server
+		ExpectedServer *webhook.Server
+	}{
+		{
+			Name:           "Wbhook server config",
+			Server:         &webhook.Server{},
+			ExpectedServer: expectedServer,
+		},
+	}
+
+	for _, tt := range tc {
+		t.Run(tt.Name, func(t *testing.T) {
+			server := congifureWebhookServer(tt.Server)
+			expectedServer.TLSOpts = server.TLSOpts
+
+			if !reflect.DeepEqual(tt.ExpectedServer, server) {
+				t.Errorf(fmt.Sprintf("got %#v, want %#v", server, tt.ExpectedServer))
+			}
+		})
+	}
+}
+
+func TestTLSConfig(t *testing.T) {
+	ca, caPEM, caPrivKey, err := getCA(*certCNName)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	certpool := x509.NewCertPool()
+	certpool.AppendCertsFromPEM(caPEM.Bytes())
+
+	serverTLSConf, err := serverCertSetup(*certCNName, ca, caPrivKey, certpool)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "success!")
+	}))
+	ts.TLS = serverTLSConf
+	ts.StartTLS()
+	defer ts.Close()
+
+	goodHTTPClient, err := getClient(*certCNName, ca, caPrivKey, certpool)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	badHTTPClient, err := getClient("test", ca, caPrivKey, certpool)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tc := []struct {
+		Name      string
+		WantError bool
+		Msg       string
+		Client    *http.Client
+	}{
+		{
+			Name:      "Connecting to server with valid certificate that has expected CN name",
```

Comment:
Can you add a test that uses a different CA but with the correct CN name?  It should fail.

`Connecting to server with invalid certificate that has expected CN name`